File: problems/0009_palindrome-number/reversed-int/Solution.py
#!/usr/bin/python3

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        elif x < 10:
            return True
        reversed, k = 0, x
        while k != 0:
            reversed = reversed * 10 + (k % 10)
            k //= 10
        return x == reversed

File: problems/0009_palindrome-number/reversed-int/Solution.cs
public class Solution {
    public bool IsPalindrome(int x) {
        if (x < 0)
            return false;
        else if (x < 10)
            return true;
        var reversed = 0;
        for (var k = x; k != 0; k /= 10)
            reversed = reversed * 10 + (k % 10);
        return x == reversed;
    }
}
